cudax/include/cuda/experimental/__container/uninitialized_buffer.h

```diff
@@ -51,20 +53,14 @@ namespace cuda::experimental
 //! need to implement :ref:`get_property(const device_buffer&, Property)
 //! <libcudacxx-extended-api-memory-resources-properties>`.
 //!
-//! .. warning::
-//!
-//!    ``uninitialized_buffer`` stores a reference to the provided memory :ref:`memory resource
-//!    <libcudacxx-extended-api-memory-resources-resource>`. It is the user's resposibility to ensure the lifetime of
-//!    the resource exceeds the lifetime of the buffer.
-//!
 //! @endrst
-//! @tparam _T the type to be stored in the buffer
+//! @tparam _Tp the type to be stored in the buffer
 //! @tparam _Properties... The properties the allocated memory satisfies
 template <class _Tp, class... _Properties>
 class uninitialized_buffer
 {
 private:
-  _CUDA_VMR::resource_ref<_Properties...> __mr_;
+  mr::any_resource<_Properties...> __mr_;
```

Comment:
Suggestion: This could potentially lead to name clashes, we should use fully qualified names
```suggestion
  ::cuda::experimental::mr::any_resource<_Properties...> __mr_;
```